conda_libmamba_solver/index.py

```diff
@@ -160,6 +162,7 @@ def __init__(
         repodata_fn: str = REPODATA_FN,
         installed_records: Iterable[PackageRecord] = (),
         pkgs_dirs: PathsType = (),
+        in_state: SolverInputState | None = None,
```

Comment:
Looks like this is not being used, is it? I'd rather pass the bits you need to the methods that require it so these three objects (Index, InputState, OutputState) do not depend on each other